src/main/java/com/ericsson/ei/services/SubscriptionService.java

```diff
@@ -61,27 +66,48 @@
     @Value("${ldap.enabled}")
     private boolean ldapEnabled;
 
+    @Value("${jasypt.encryptor.password:}")
+    private String jasyptEncryptorPassword;
+
     @Autowired
     private ISubscriptionRepository subscriptionRepository;
 
     @Override
-    public void addSubscription(Subscription subscription) throws JsonProcessingException, MongoWriteException {
+    public void addSubscription(Subscription subscription)
+            throws JsonProcessingException, MongoWriteException {
+        String authType = subscription.getAuthenticationType();
+        String username = subscription.getUserName();
+        String password = subscription.getPassword();
+        boolean authenticationDetailsProvided = isAuthenticationDetailsProvided(authType, username,
+                password);
+        if (authenticationDetailsProvided && !StringUtils.isEmpty(jasyptEncryptorPassword)
+                && !AuthenticationType.valueOf(authType).equals(AuthenticationType.NO_AUTH)) {
+            StandardPBEStringEncryptor encryptor = new StandardPBEStringEncryptor();
+            encryptor.setPassword(jasyptEncryptorPassword);
+            String encryptedPassword = String.format(ENCODED_PASSWORD, encryptor.encrypt(password));
```

Comment:
ENCODED_PASSWORD is not an encoded password. Please find a better name.

Reply:
I now call it ENCODED_WRAPPER